internal/publicip/api/errors.go

```diff
@@ -6,4 +6,5 @@ var (
 	ErrTokenNotValid   = errors.New("token is not valid")
 	ErrTooManyRequests = errors.New("too many requests sent for this month")
 	ErrBadHTTPStatus   = errors.New("bad HTTP status received")
+	ErrUnsupportedIP   = errors.New("service does not support arbitrary IP addresses")
```

Comment:
nit if I may, perhaps this is more appropriate:

```go
ErrServiceLimited = errors.New("service is limited")
```

and then wrap it in the cloudflare code with `fmt.Errorf("%w: cloudflare cannot provide information on the arbitrary IP address %s", ErrServiceLimited, ip)` - that way that 'common' error `ErrServiceLimited` can also be used elsewhere in the future. It also feels a bit wrong that it's "unsupported ip", it would be supported if it would be our IP address, so it's more of a service limitation 🤔